### ISSUE TITLE
@RegisterWidget throw exception when we use concrete types like `ArrayList` as field.

### ISSUE THREAD

**Issue by othiagosouto** [ZupIT/beagle#715]
## Description
@RegisterWidget throw exception when we use concrete types like `ArrayList` as field even when we don't use `moshi`.  This 

java.lang.IllegalArgumentException: Platform class java.util.ArrayList in java.util.ArrayList<com.project.someview.viewpage.ViewPageData> requires explicit JsonAdapter to be registered

## Steps To Reproduce
Provide a detailed list of steps that reproduce the issue.

1. Create a WidgetView and use @RegisterWidget
2. Use a constructor field like ArrayList.

## Expected Results
Should not throw exception

## Code example, screenshot, or link to a repository:
Create a component with concrete type.
```
@RegisterWidget
class SomeViewWidget
    (
    private var txtBtnSkip: String,
    private var txtBtnContinue: String,
    private var data: ArrayList<ViewPageData> = ArrayList(),
    private var path: String
) : WidgetView() {
    override fun buildView(rootView: RootView): View {
        return SomeView(context = rootView.getContext())
    }
}
```

**Comment by tuliopereirazup** [comment]
Hello @othiagosouto, we will create a section in our doc explaining these cases and in the next version will be possible to manipulate our internal moshi instance.

**Comment by othiagosouto** [comment]
hey @tuliojuniorzup  the same thing happens to `CharSequence` and `Pair`.

How the next version will handle this? The application will need to know details about moshi? If so, this may lead to some bad practice since you are exposing details of framework to external clients (applications) and forcing some transitive dependency.

**Comment by uziasferreirazup** [comment]
Sorry about delay to answer you, the developer team will purpose some solution in the next days, probably will create some protocol to developer implements the parse of your objects.

**Comment by diogooliveira-zup** [comment]
Hey @othiagosouto we will implement a improvement that you will create your own adapter when moshi could not deserialize your attribute, in this case you will override the deserialization/serialization by your own deserialization adapter, cya!

**Comment by othiagosouto** [comment]
Hey @diogooliveira-zup , thanks for answer me.

Do we have any forecast date to provide this? When 1.3 version will be available?

**Comment by uziasferreirazup** [comment]
@othiagosouto in the next release will be available this feature.

Example of the interface and annotation beagle will be provided in the next version:

```
private const val KEY = "first"
private const val VALUE = "second"

@RegisterBeagleAdapter
class PairAdapter : BeagleTypeAdapter<Pair<String, String>> {

    override fun fromJson(json: String): Pair<String, String> {
        val rootObject = JSONObject(json)
        return Pair(rootObject.getString(KEY), rootObject.getString(VALUE))
    }

    override fun toJson(type: Pair<String, String>): String {
        val rootObject = JSONObject()
        rootObject.put(KEY, type.first)
        rootObject.put(VALUE, type.second)
        return rootObject.toString()
    }
```

the documentation of this interface and annotation will be available at the doc.

**Comment by uziasferreirazup** [comment]
@othiagosouto this feature will be available in version 1.2.2-android and will be generated this week.
I will close this issue because was merged in the branch of the release, but if you have any doubts, feel free to open another issue.
